agent/util-scripts/pbench-register-tool

```diff
@@ -222,54 +234,52 @@ fi
 
 if [[ ${remotes_arg::1} == "@" ]]; then
 	# We are dealing with a file containing the list of remotes.
+	remotes_file=${remotes_arg:1}
 
 	if [[ -n ${labels_arg} ]]; then
 		error_log "--labels=${labels_arg} not allowed with remotes file (${remotes_arg})"
 		usage >&2
 		exit 1
 	fi
-	remotes_file=${remotes_arg#*@}
-	if [[ ! -e ${remotes_file} ]]; then
-		error_log "--remotes=@${remotes_file} specifies a file that does not exist"
+
+	if [[ ! -r ${remotes_file} ]]; then
+		error_log "--remotes=@${remotes_file} specifies a file that does not exist or is not readable"
 		usage >&2
 		exit 1
 	fi
+
 	declare -a remotes_A
 	declare -A labels_A
 	idx=0
 	lineno=0
 	while read line; do
-		if [[ ${line::1} == "#" ]]; then
+		(( lineno++ ))
+		# a comment line is in regular expression form:  [ \t]*#.*\n
+		if [[ ${line} == *([[:space:]])#* ]]; then
 			# Ignore comments
-			(( lineno++ ))
 			continue
 		fi
 		IFS=',' read -ra parts <<< "${line}"
-		if [[ ${#parts[@]} == 0 ]]; then
-			# Ignore empty lines
-			(( lineno++ ))
+		# we need arithmetic evaluation of the conditionals
+		if (( ${#parts[@]} == 0 )); then
+			# Ignore empty lines: [ \t]*\n
```

Comment:
Again, this regex is a bit misleading since you're actually relying on bash tokenization through `read -a` and just checking for "no tokens parsed".

Reply:
Yeah, I punted on that and deleted the regex: I'd need to check precisely what the parsing does and probably add test cases - mañana...